packages/replay-next/components/windowing/GenericListData.ts

```diff
@@ -78,8 +78,10 @@ export abstract class GenericListData<Item> extends EventEmitter<{
   };
 
   setSelectedIndex(value: number | null) {
-    this._selectedIndex = value;
-    this.emit("selectedIndex", value);
+    if (this._selectedIndex !== value) {
+      this._selectedIndex = value;
+      this.emit("selectedIndex", value);
+    }
```

Comment:
We should land this change regardless. Without it we have a potential update loop between this and `useSyncExternalStore`.